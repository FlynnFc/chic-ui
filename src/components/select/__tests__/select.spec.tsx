import Select from '..';
import { render } from '../../../test/utils';

describe('Select Component', () => {
  it('Renders Correctly', () => {
    const SelectGroup = render(
      <>
        <div>
          <Select
            width="200px"
            data={['India', 'Japan', 'Germany', 'Russia']}
            placeholder="Select a country"
            onChange={(value) => alert(value)}
          />
        </div>
        <div>
          <Select
            multiSelect
            width="400px"
            data={[
              { id: 1, game: 'Hockey' },
              { id: 2, game: 'Football' },
              { id: 3, game: 'Tennis' }
            ]}
            dataKey="id"
            dataName="game"
            placeholder="Select a sport"
            onChange={(value) => alert(value)}
          />
        </div>
      </>
    );
    expect(SelectGroup).toMatchSnapshot();
  });
});
