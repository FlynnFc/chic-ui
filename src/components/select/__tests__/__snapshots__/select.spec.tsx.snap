// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component Renders Correctly 1`] = `
Array [
  <div>
    <div
      className="sc-bdfBQB iUommc"
      width="200px"
    >
      <div
        className="sc-dlfnuX hdSKUI"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
      >
        <div>
          Select a country
        </div>
        <svg
          className="sc-eCstlR iGbiJz"
          viewBox="0 0 330 330"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M325.607 79.393c-5.857-5.857-15.355-5.858-21.213.001l-139.39 139.393L25.607 79.393c-5.857-5.857-15.355-5.858-21.213.001-5.858 5.858-5.858 15.355 0 21.213l150.004 150a14.999 14.999 0 0021.212-.001l149.996-150c5.859-5.857 5.859-15.355.001-21.213z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="sc-bdfBQB dXXAAm"
      width="400px"
    >
      <div
        className="sc-dlfnuX hdSKUI"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
      >
        <div>
          Select a sport
        </div>
        <svg
          className="sc-eCstlR iGbiJz"
          viewBox="0 0 330 330"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M325.607 79.393c-5.857-5.857-15.355-5.858-21.213.001l-139.39 139.393L25.607 79.393c-5.857-5.857-15.355-5.858-21.213.001-5.858 5.858-5.858 15.355 0 21.213l150.004 150a14.999 14.999 0 0021.212-.001l149.996-150c5.859-5.857 5.859-15.355.001-21.213z"
          />
        </svg>
      </div>
    </div>
  </div>,
]
`;
